content/articles/Integrating-mongodb-with -java/index.md

```diff
@@ -0,0 +1,241 @@
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+```
+To connect to a running MongoDB instance, use MongoClient() inside the `public static void main` as shown below.The MongoClient instance represents a pool of database connections; even if you have multiple threads, you only need one instance of the MongoClient class.
+
+```java 
+// Creating a Mongo client
+public static void main(String[] args) throws UnknownHostException {MongoClient mongoClient= new MongoClient( "localhost", 27017);//connecting to the database
+MongoDatabase db=mongoClient.getDatabase("SchoolManagementSystem");
+```
+### Creating MongoDB collection
+MongoDB stores data records as documents that are organized into collections.
+To create a collection, type the code below after the created mongo client.
+
+```
+MongoCollection> collection<Document>collection=db.getCollection("students");
+```
+it should look like this
+
+```java
+import java.net.UnknownHostException; //Its used if the IP address of a hostname could not be determined
+import java.util.Iterator; //an Iterator which is used to loop through collections is imported from java.util package
+import org.bson.Document;// Parses a string in MongoDB Extended JSON format to a Document
+import com.mongodb.*;
+import com.mongodb.DB;
+import com.mongodb.DBCollection;
+import com.mongodb.DBCursor;
+import com.mongodb.DBObject;
+import com.mongodb.MongoClient;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection; import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+
+public static void main(String[] args) throws UnknownHostException {MongoClient mongoClient= new MongoClient( "localhost", 27017);//connecting to the database
+MongoDatabase db=mongoClient.getDatabase("SchoolManagementSystem");
+
+MongoCollection> collection<Document>collection=db.getCollection("students");
+}
+
+```
+ 
+To confirm if the collection has been created, type show collections in the cmd terminal.
+
+![Created Collections](/engineering-education/integrating-mongodb-with-java/created-collections.png)
+
+### Insert data into the collection
+
+Add the following code to your Java class after the created collection.
+```java
+//inserting a new document in the collection which will be student number one.
+Document doc=new Document("name","Christine Muthoni");doc.append("id", "CT201/100091/20");doc.append("course", "Clinical Medicine");doc.append("Gender", "Female");collection.insertOne(doc);System.out.println("Insert is completed");
+
+//insert a new document in the collection, which will be student number two. 
+Document doc2=new Document("name", "Esther Wacera");doc2.append("id", "CT201/10999/14");doc2.append("course", "Civil Engineering");doc2.append("Gender", "Female");collection.insertOne(doc2);System.out.println("Insert is completed");
+```
+In the cmd terminal type
+``` 
+myCursor=db.students.find({})
+```

```

Comment:
Are you sure you type this in the terminal?

Reply:
hey @srishilesh  
Yes i'm sure,,